### DISCUSSION TITLE
Any way to prevent Pydantic from treating programmer mistakes as validation errors?

### DISCUSSION THREAD

**Discussion by verngutz** [pydantic/pydantic#6254] [Question] [answered]
Every now and then, I make a bug in my validator, which unintentionally results in a TypeError or ValueError. For example, maybe I forgot to check if an optional list field was None before iterating over it. Particularly when this happens in a root validator, it is difficult to debug, as the error turns into a validation error and I get a not-very-helpful message like:

```
pydantic.error_wrappers.ValidationError: 1 validation error for Model
__root__
  'NoneType' object is not iterable (type=type_error)
```

Is there some way to prevent Pydantic from treating programmer mistakes as validation errors? I'd much rather have this throw a normal TypeError and see the stack trace so that I can find the offending line in my validator.

(currently on pydantic version 1.9.0)

**Comment by samuelcolvin** [comment, marked as answer]
I agree this is a problem in pydantic v1, it's changed in pydantic V2 where only `ValueError` and `AssertionError` cause a `ValidationError`.
